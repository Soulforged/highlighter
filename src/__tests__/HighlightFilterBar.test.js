import React from 'react';
import {
  render,
  fireEvent,
  cleanup,
  waitForElement
} from "@testing-library/react";
import {
  HighlightsFilterBar
} from '../components';

describe("highlight filter bar tests", () => {

  afterEach(cleanup);

  it("renders filters with selections", () => {
    const { container, getByTestId } = render(<HighlightsFilterBar filter={["yellow"]} />);
    const selectedOption = getByTestId("yellow");
    const notSelectedOption = getByTestId("red");
    expect(selectedOption.style.border).toEqual("3px solid blue");
    expect(notSelectedOption.style.border).toEqual("3px solid black");
    expect(container).toMatchSnapshot();
  });

  it("toggles filter selection on filter option click", () => {
    const removeSelectedHighlightOption = jest.fn();
    const addSelectedHighlightOption = jest.fn();
    const { getByTestId } = render((
      <HighlightsFilterBar
        filter={["yellow"]}
        removeSelectedHighlightOption={removeSelectedHighlightOption}
        addSelectedHighlightOption={addSelectedHighlightOption}
      />
    ));
    const selectedOption = getByTestId("yellow");
    const notSelectedOption = getByTestId("red");
    fireEvent.click(selectedOption);
    expect(removeSelectedHighlightOption).toHaveBeenCalled();
    expect(addSelectedHighlightOption).not.toHaveBeenCalled();
    fireEvent.click(notSelectedOption);
    expect(addSelectedHighlightOption).toHaveBeenCalled();
  });
});
