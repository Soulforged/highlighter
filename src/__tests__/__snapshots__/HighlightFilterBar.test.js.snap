// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`highlight filter bar tests renders filters with selections 1`] = `
<div>
  <div
    style="flex-direction: row; padding: 10px;"
  >
    <button
      data-testid="red"
      style="width: 50px; height: 32px; margin: 5px; background-color: red; border: 3px solid black;"
    />
    <button
      data-testid="yellow"
      style="width: 50px; height: 32px; margin: 5px; background-color: yellow; border: 3px solid blue;"
    />
    <button
      data-testid="green"
      style="width: 50px; height: 32px; margin: 5px; background-color: green; border: 3px solid black;"
    />
  </div>
</div>
`;
